cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -1541,6 +1543,55 @@ genBoundsCommand =
         ]
     }
 
+-- ------------------------------------------------------------
+-- Check command
+-- ------------------------------------------------------------
+
+data CheckFlags = CheckFlags
+  { checkVerbosity :: Flag Verbosity
+  , checkIgnore :: [CheckExplanationIDString]
+  }
+  deriving (Show, Generic, Typeable)
```

Comment:
Does it have to be `Generic`? Just your regular reminder that `Generic` is not free and can cost sensible compilation time.

Reply:
No need indeed, eliminated.